### PR TITLE
two endpoints added to get and search  one trainee

### LINKED ISSUE #30: Boshra API endpoint to get one username details
as a developer I want to get all the data for codewars and github for a trainee so we can use it in front end to give information to a trainee or a product manger

### PR DESCRIPTION
**Pull request by BoshraM**
I am not sure if the way I write these  two endpoint separately is good or not . we  need to use search endpoint when the trainees search for their milestone , I haven't connect this end point with data base ? may I add data base as well so when trainee see their dashboard we can  add some information from data base to dashboard for example which cohort they or which full name they have been registered?
we  need to use get endpoint with id  when product manager wants to view trainee information as I am not sure yet what we want to show them in this point ? 

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:
related #30 
Fixes #30 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [x] I have updated any documentation

### CODE REVIEW

File: server/controllers/traineeController.js [hunk 1]
```diff
@@ -1,11 +1,55 @@
...
+        const trainee = result.rows[0];
+
+        if (!trainee) {
+            return res.status(404).json({ error: "Trainee not found" });
+        }
+
+        res.status(200).json({ trainee });
+    } catch (error) {
+        console.error("Error retrieving trainee:", error);
+        res.status(500).send("Failed, Internal Server Error. Database or table not found");
```
Comment (PeteLindsell) [inline server/controllers/traineeController.js:22]: Probably no need to tell the API user 'Database or table not' as there isnt anything they can do about it

File: server/controllers/traineeController.js [hunk 2]
```diff
@@ -1,11 +1,55 @@
...
+        const githubData = await githubDataResponse.json();
+
+        res.status(200).json({
+            codewarsData,
+            githubData,
+        });
+    } catch (err) {
+        console.error(err);
+        res.status(500).json("Failed, Internal Server Error");
+    }
```
Comment (PeteLindsell) [inline server/controllers/traineeController.js:49]: I was thinking that this endpoint would get the trainee from our database returning just there name and cohort for now.
You could extend the trainee/:id endpoint to allow gitHubUserName or id. But having this search endpoint could be useful moving forward

**Review by PeteLindsell — COMMENTED** [review]
A couple of comments. Not sure if we get any benefit by calling github and codewars from the backend here.

File: server/controllers/traineeController.js (hunk 1 above)
Comment (BoshraM) [reply server/controllers/traineeController.js:22]: thank you, i just copied so with mistake : ), hev ebeen removed

**Review by PeteLindsell — APPROVED** [review]
Looks really good 👍

**Pull request merged**